composer.json

```diff
@@ -1,15 +1,51 @@
 {
-  "name": "i-Educar",
+  "name": "portabilis/i-educar",
   "description": "Software livre de gestão escolar",
   "type": "project",
   "license": "GPL-2.0-or-later",
-  "config": {
-    "vendor-dir": "ieducar/vendor"
-  },
+  "keywords": [
+    "Portabilis",
+    "Ieducar"
+  ],
+  "homepage": "https://github.com/portabilis/i-educar",
   "require": {
+    "php": "~7.0",
     "portabilis/jasperphp": "v1.2.0",
     "robmorgan/phinx": "v0.8.1",
     "gilbitron/php-simplecache": "^1.4",
     "cocur/slugify": "^3.1"
+  },
+  "require-dev": {
+    "phpunit/phpunit": "~7.0"
+  },
+  "suggest": {
+    "fabpot/php-cs-fixer": "Fix PSR2 and other coding style issues"
+  },
+  "autoload": {
```

Comment:
Não vejo **super** necessidade da gente já definir os autoloaders pro projeto já que ele não tem nada em PSR-4 ainda... Mas, se for pra definir eu acho que ficaria menos verboso usarmos apenas `Ieducar` pro projeto principal e `Ieducar\Test` pra testes.

Reply:
Concordo em usar o autoload apenas para dev, mas o Namespace é desta forma `\\Portabilis\\Ieducar\\` pela convenção da PRS-4 `\<NamespaceName>(\<SubNamespaceNames>)*\<ClassName>` Poderiamos trabalhar em algo como `Ieducar\Foo` se tudo dentro de src fosse separado por módulos. Hoje tem lib, module, modules. Acredito que no momento onde estivermos removendo todos os `require_once` e migrando para namespace conseguiremos trabalhar em uma organização melhor.

Reply:
Aaaa e testes não devem ter uma namespace de teste como o exemplo `Ieducar\Test` que você deu, deve ser a mesma namespace da aplicação para você executar os testes no mesmo nível.

Reply:
Faz sentido @williamespindola ;)

Reply:
@eberfreitas fix o teste aqui e não tem problema deixar o autoload, ele não esta usando.

Reply:
@williamespindola  só sobre os namespaces... Pelo que vi nada impede que o nosso NamespaceName seja efetivamente `\Ieducar` já que um SubNamespaceName não é obrigatório. Talvez essa ridigez fizesse mais sentido com o PSR-0 que está depreciado, mas com o PSR-4 a gente informa o namespace e define um um diretório base arbitrário qualquer.

Justamente pelo código hoje não ter uma organização, também acho que seria legal irmos movendo as classes para fora da pasta `ieducar` e para dentro de uma `src` conforme elas fossem se adequando ao PSR-4.

Reply:
@eberfreitas eu dei uma lida la e cafundi tudo, nenhuma das duas é obrigatório o sub na verdade. Valeu \o/